### ISSUE TITLE
no providers for meta.config

### ISSUE THREAD

**Issue by bergben** [vinaygopinath/ng2-meta#1]
Hey there,

first of all nice work. Wanted to ask if RC4 is supported? 
I am getting 
`browser_adapter.js:84ORIGINAL EXCEPTION: No provider for meta.config!` :(

this is my app.ts

``` javascript
import "reflect-metadata";
import "zone.js/dist/zone";
import { Component } from "@angular/core";
import { bootstrap } from "@angular/platform-browser-dynamic";
import { appRouterProviders } from "./app.routes";
import { ROUTER_DIRECTIVES } from "@angular/router";
import { Title } from '@angular/platform-browser';
import template from "./app.html";
import { MetaService } from "ng2-meta";

@Component({
  selector: "app",
  template,
  directives: [ROUTER_DIRECTIVES],
  providers: [MetaService]
})

export class app {
  constructor(private metaService: MetaService) {}
}

bootstrap(app, [
  appRouterProviders,
  Title,
  MetaService
])
.catch(err => console.error(err));
```

routes: 

``` javascript
const routes: RouterConfig = [
  { path: "blog", component: Blog,
        children:[
            { path: "", component: BlogIndex },
            { path: "**", redirectTo: "Blog" },
        ]
  },
  { path: "", component: About,
       data: {
          meta: {
            title: 'About',
            description: 'Description of about'
          }
        },
        children:[
            { path: "", component: AboutIndex },
            { path: "**", redirectTo: "About" }
        ]
  },
  { path: "**", redirectTo: "About" }
];

export const appRouterProviders = [
  provideRouter(routes)
];
```

**Comment by vinaygopinath** [comment]
Thanks for using ng2-meta and creating an issue! Sorry for the delayed response - I'm on a laptop-free vacation at the moment and GitHub on mobile is annoying (Why isn't there a GitHub mobile app!)

RC4 is indeed supported. Your issue seems to be related to the additional configuration that can be provided to MetaService during the bootstrap process. I meant to make the "meta.config" configuration optional, but it looks like I've forgotten to add the @Optional annotation in the [code](https://github.com/vinaygopinath/ng2-meta/blob/master/src/meta.service.ts#L29). I'll check if this is the cause of your issue and release an update later this week.

If you need a quick workaround, I suggest adding a "meta.config" provider, as shown in the [README](https://github.com/vinaygopinath/ng2-meta/blob/master/README.MD#set-defaults) and this [bit of code](https://github.com/vinaygopinath/NomadCouple/blob/master/src/main.ts#L14-L27)

**Comment by bergben** [comment]
Hey there! Don't worry about the delayed response, enjoy your vacation! 
Thank you for the answer, it was straight to the point. I have tested it and it works perfectly. I have created a pull request with the few changes in case you want to merge it. :)

**Comment by vinaygopinath** [comment]
Thanks for your help :) Just merged your fix.

I still need to release a new version to update npm, so I'll leave this issue open until then.

**Comment by vinaygopinath** [comment]
Fixed in [v1.0.0](https://github.com/vinaygopinath/ng2-meta/releases/tag/v1.0.0)

**Comment by sanjaykumarns** [comment]
So which version I have to use in package.json in order to avoid this "No provider for Token meta config!" error. I have used 1.0.0, Mine was, 2.0.3 but in both case, It didn't worked. Can you help?

I tried as you suggested in the Set defaults section of the above link, But then also I am getting the following error:

ERROR in Error encountered resolving symbol values statically. Calling function 'MetaModule', function calls are not supported. Consider replacing the function or lambda with a reference to an exported function, resolving symbol GeneralModule in /var/www/SSNW/client/src/app/general/general.module.ts, resolving symbol GeneralModule in /var/www/SSNW/client/src/app/general/general.module.ts
